### PR TITLE
Release of version 0.13.2

### PR DESCRIPTION
**Pull request by paulromano**
To support the [OpenMC course](https://www.oecd-nea.org/jcms/pl_72490/openmc-online-course) we're organizing next week, @pshriwise @shimwell and I would like to put out a quick release that incorporates the latest bug fixes and enhancements. There haven't been many breaking API changes, so I thought it made more sense to change the version number to 0.13.2 (rather than 0.14.x).

To summarize the changes here:
- Update version numbers
- Fixed some documentation build warnings
- Fixed a build warning when using clang

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -69,9 +69,9 @@
 # built documents.
 #
 # The short X.Y version.
-version = "0.14"
+version = "0.13"
```
Comment (shimwell) [inline docs/source/conf.py:72]:
```suggestion
import openmc
version = '.'.join(openmc.__version__.split('.')[0:2])
```

File: docs/source/conf.py [hunk 2]
```diff
@@ -69,9 +69,9 @@
...
 # The full version, including alpha/beta/rc tags.
-release = "0.14.0"
+release = "0.13.2"
```
Comment (shimwell) [inline docs/source/conf.py:74]: Same as above, I'm just wondering if this version can be found from the package automatically
```suggestion
release = openmc.__version__
```

File: docs/source/conf.py (hunk 1 above)
Comment (paulromano) [reply docs/source/conf.py:72]: I'm not sure importing the package within conf.py will work smoothly, so I'd rather not take a chance at messing up our documentation on release. We can try this out in the future though!

**Review by shimwell — APPROVED** [review]
Looks good to me

**Review by pshriwise — APPROVED** [review]
LGTM! Thanks @paulromano 😄

**Pull request merged**